app/forms/Mapping.scala

```diff
@@ -20,16 +20,26 @@ import play.api.data.format.Formatter
 import play.api.data.{FieldMapping, FormError}
 
 object Mapping {
-
   def requiredRadio(requiredKey: String = "error.required"): FieldMapping[String] =
     of(radioFormatter(requiredKey))
 
-  private def radioFormatter(requiredKey: String): Formatter[String] = new Formatter[String] {
+  def requiredRadio(requiredKey: String, choices: Seq[String]): FieldMapping[String] =
+    of(radioFormatter(requiredKey, choices))
+
+  def optionalRadio(requiredKey: String = "error.required", choices: Seq[String]): FieldMapping[String] =
```

Comment:
Nit: Just realized that `optionalRadio` and `"error.required"` does not seem right here. Probably we need to think about more apropriate naming.